### ISSUE TITLE
Feature Request:  Add "When Guard" ability to Checks

### ISSUE THREAD

**Issue by Exocomp** [arachnys/cabot#460]
Hi,

I have not seen this feature and have use cases for it so would be nice to have.

**Check**: SQL Page Lookups Per Sec Warning
**Metric 1**: SQL Page Lookups per sec
**Metric 2**: Batch Requests per sec
**Overall Metric**: diffSeries of Metric 1 & Metric 2

I only want the check to continue if Metric 2 is above a certain threshold.  Why? Well because it doesn't make sense when the SQL server is not busy.

If there was a "When Guard" option (to specify when to trigger) that would solve the problem.

Or if there was a check that got promoted to another check.  So overall check only runs if check 1 is promoted.  Something like that.  Hope that makes sense.

Thanks,
E

**Comment by JeanFred** [comment]
Hi,

Here again, I think Graphite functions will give you what you what. If you apply a [removeBelowValue](https://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.removeBelowValue) together with a transformNull to metric2, as in `transformNull(removeBelowValue(metric1, threshold), 0)` (or smth to that effect) then you can give an arbitrary value to metric2 which ensures that your check does not trigger unless the threshold is met.

I can understand that it may not be pretty, but on the other hand part of the point of using Graphite as a backend is to take advantage of its calculations capabilities to do all the heavy lifting. :)

(I am not sure I understand your use-case perfectly either which makes it hard for me to make suggestions :-/)

**Comment by Exocomp** [comment]
@JeanFred I can live with it being ugly as long as it works, :), I will give you solution a try and will follow up.  Thanks!!

**Comment by Exocomp** [comment]
@JeanFred 

I made some progress using your suggestion but still not there yet.

**Overall Metric**: diffSeries of Metric 1 & Metric 2 has to be below zero (if above trigger)

So using your suggested solution:

Metric 2:  transformNull(removeBelowValue(metric2, **10**), 0)

Overall query:

`diffSeries(Metric1, transformNull(removeBelowValue(metric2, 10), 0))`

Problem here is that for the results where metric2 is zero when subtracted from metric1 it doesn't really produce a value that is actionable.  However if I set it to a value that is off the scale something like 10000:

`diffSeries(Metric1, transformNull(removeBelowValue(metric2, 10), 10000))`

Then the diffSeries produces values in the negatives for the cases I don't care about which something very clear to ignore.

So this looked promising, but always the first value of the diffSeries is not calculated properly (it is always 10k - what ever I set it to with transformNull), so in Cabot it does not work out to where I can say only alert when above 0.
